azavea/cac-tripplanner
# Add carousel for Tour cards

## Overview

What this does so far:
- Adds an image carousel to the Tour cards on the home page and the Explore view.  It uses the main image from each destination.
- Since I was adding two locations where the image carousel needed to be initialized, and there were already a couple, and the options are mostly the same across the various invocations (and it seems unlikely that we'd want to change them independently of each other), I split most of the options for the "tiny slider" function into a shared object in `utils.js`.
- The Tour cards weren't showing up on the Explore view, even when I selected "Tour" from the filter list, so I adjusted the style to make them show up there but still hide when "All" is selected.

What this doesn't do yet:
- The other task from [the issue](https://github.com/azavea/cac-tripplanner/issues/1146) is "'Directions' card link should be labelled 'Map' instead for (non-routable) event map overview".  This is the same thing as the [link on the Tour Detail page](https://github.com/azavea/cac-tripplanner/blob/1616d5506af6dd2371672c019feb2a43b48281a1/python/cac_tripplanner/templates/partials/tour-detail.html#L28-L31), but there's currently only a back-end function for composing it.  To get it working in the front-end version of the templates, we need to either write a front-end tag to reproduce the behavior or stuff that link into the API response.  I was leaning toward the latter, since it's a fairly involved bit of business, but I'm wondering what you think would be best.

### Demo

![tour_carousels](https://user-images.githubusercontent.com/6598836/66149449-2143bb00-e5e1-11e9-8081-1f86e62a278f.gif)

### Notes

The carousel images are coming out slightly narrower than the normal images, resulting in some dead space on each side in the cards.  The cards are also slightly taller, due to the row of dots that the carousel adds.  Since there's a styling pass planned for the near future, I didn't pursue the matter...

## Testing Instructions

- Load the home page. The tour cards should show up with image carousels rotating through the main images from each of the tour destinations (once each, in order).
- Filter to "Tours". Now you're seeing the version of the cards rendered by Javascript. They should look the same.
- Click over to the map view.  The cards should still be there, and still have working carousels.

## Checklist

Resolves #1146.
